merino/jobs/wikipedia_indexer/filemanager.py

```diff
@@ -0,0 +1,142 @@
+"""Manage files between wikimedia exports and gcs bucket"""
+import logging
+import re
+from datetime import datetime as dt
+from gzip import GzipFile
+from html.parser import HTMLParser
+from typing import Generator, List, Optional, Pattern, Tuple
+from urllib.parse import urljoin
+
+import requests
+from google.cloud.storage import Blob, Client
+from google.cloud.storage.fileio import BlobReader, BlobWriter
+
+logger = logging.getLogger(__name__)
+
+
+class DirectoryParser(HTMLParser):
+    """Parse the directory listing to find the specified file."""
+
+    filter: Pattern[str]
+    file_paths: List[str] = []
+
+    def __init__(self, filter_wildcard: Pattern[str]) -> None:
+        super().__init__()
+        self.filter = re.compile(filter_wildcard)
+
+    def _is_href(self, k: str, v: str) -> bool:
+        return k == "href" and re.search(self.filter, v) is not None
+
+    def handle_starttag(self, tag: str, attrs: List[Tuple[str, Optional[str]]]) -> None:
+        """When the parser encounters a start tag check for Anchor and push into list."""
+        if tag.lower() == "a":
```

Comment:
Per [API doc](https://docs.python.org/3/library/html.parser.html#html.parser.HTMLParser.handle_starttag), `tag` should already be lower case.
```suggestion
        if tag == "a":
```